### PR TITLE
Make update and create conform to json-api resource structure

### LINKED ISSUE #24: Process Update and Create responses through json-api formatter
Right now we're sending data straight through as the response. This should also be processed through the formatter, as this data has the same requirements as a read result.

Will require rewriting some unit tests and verifying sensible results.

### PR DESCRIPTION
**Pull request by bobholt**
- move request and response data under "data" key instead of type key
- parse update and create responses through json-formatter
- update unit tests for create and update to use real models
- add integration test for update because i needed it to find a bug in the
unit tests

fixes #24

### CODE REVIEW

File: modules/formatter-jsonapi/index.js [hunk 1]
```diff
@@ -4,83 +4,93 @@ const getKey = require('./lib/get_key');
 const toOneRelations = require('./lib/to_one_relations');
 const link = require('./lib/link');
 
+var linkWithInclude;
+var linkedResourceIndex;
+var typeName;
+
+function formatModel(output, model) {
```
Comment (tkellen) [inline modules/formatter-jsonapi/index.js]: can we move this into `lib/format_model.js`?

File: modules/formatter-jsonapi/index.js [hunk 2]
```diff
@@ -4,83 +4,93 @@ const getKey = require('./lib/get_key');
...
+  var linkWithoutInclude = _.difference(Object.keys(toOneRels), linkWithInclude);
+  // get the or initialize the primary resource key
+  var primaryResource = getKey(output, 'data');
+  // get a json representation of the model, excluding any related data
+  var serialized = model.toJSON({shallow:true});
+  // build a links object, adding any linked models to
+  var links = link(model, {
+    linkWithInclude: linkWithInclude,
+    linkWithoutInclude: linkWithoutInclude,
+    exporter: function (models, type) {
```
Comment (tkellen) [inline modules/formatter-jsonapi/index.js]: this will probably have to become a param to `lib/format_model.js`

**Comment by tkellen** [comment]
:+1: other than getting that stateless method into it's own file and unit tested alone.

**Comment by bobholt** [comment]
moved formatModel to its own file and added a unit test

this makes the first unit test in the create and update responders even less relevant - may remove those when we get full integration coverage

**Comment by tkellen** [comment]
:tada:

**Comment by tkellen** [comment]
Love it! merge away

**Pull request merged**